openedx-unsupported/ecommerce
# feat: Add cookie monitoring middleware

## Description

Adds CookieMonitoringMiddleware so we can get observability into cookies
we read and set, and into cookie sizes or other operational issues. We're
trying to add this to all of the Python IDAs.

This was the smallest set of requirements changes I could make. Automated
upgrades have been broken for 3 months, so I don't want to bring in other
changes. (FYI: This unbreaks the first step of `make upgrade` by upgrading
pip-tools, but there are still version conflicts that prevent it from
completing.)

Requirements change construction:

- Manually edited pip_tools.txt to use 6.6.0
- Temporarily removed `--upgrade` from `make upgrade` calls
- Temporarily changed `edx-django-utils` to `edx-django-utils>=4.8.1` in
  base.in
- Ran `make upgrade`
- Removed temporary changes

(This is a combination of the "compile-requirements" approach used in
edx-platform and the "upgrade just one dependency" instructions. It should
produce a self-consistent output.)

ref: ARCHBOM-2072

## Testing instructions

After making appropriate remote-config changes, send in a request with a large cookie (above configured threshold) and ensure a message is logged. ecommerce should also move to the "update complete" column in the https://onenr.io/0qQapPWDpj1 dashboard (2U internal) as cookie metrics start being reported.
